### PR TITLE
[Feature] Support gen hidden states use sglang with fp8 quant.

### PR DESCRIPTION
**Pull request by jiapingW**
## Motivation
To improve inference performance, FP8 quantization is often used for inference. Therefore, is it possible to directly use the hidden states generated by FP8 quantization for training the draft model? This could also improve the speed of hidden state generation. This PR is currently under testing, and I will add some relevant performance results later. Everyone is welcome to test it.

## Related Issues

https://github.com/sgl-project/SpecForge/issues/294
## Checklist

- [✅] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

**Comment by FrankLeeeee** [comment]
I have collated the sglang-related args in this PR #317 , perhaps this PR can be directed to the `feature/sglang-args` branch instead.

**Comment by jiapingW** [comment]
> I have collated the sglang-related args in this PR #317 , perhaps this PR can be directed to the `feature/sglang-args` branch instead.

I think is OK. I'll handle it tomorrow. The quant response is listed below using 1w data and train 2 epochs. The numer is the TPOT. 

In the training set,

| inference\train | with quant | wo quant |
| :--- | :--- | :--- |
| **with quant** | 5.5 | 6.07 |
| **wo quant** | 4.51 | 4.98 |

In the test set,

| inference\train | with quant | wo quant |
| :--- | :--- | :--- |
| **with quant** | 7.11 | 7.10 |
| **wo quant** | 5.93 | 5.85 |

Based on my results, quantification doesn't seem to have a significant decrease on the outcome. But use quant can achieve 50% training speedup in H20.

**Comment by jiapingW** [comment]
@FrankLeeeee I have refatored it.

File: scripts/prepare_hidden_states.py [hunk 1]
```diff
@@ -125,14 +127,15 @@ def build_target_model(
...
     else:
+        target_model_kwargs = SGLangBackendArgs.from_args(args).to_kwargs()
         target_model = get_eagle3_target_model(
             pretrained_model_name_or_path=args.target_model_path,
             backend="sglang",  # we set this as the default backend to minimize precision mismatch in training and serving
             torch_dtype=torch.bfloat16,
             device="cuda",
             cache_dir=args.cache_dir,
+            **target_model_kwargs,
         )
```
Comment (FrankLeeeee) [inline scripts/prepare_hidden_states.py]: The `torch_dtype` is hardcoded as bfloat16, I assume this will cause a conflict with the user args.

**Review by FrankLeeeee — APPROVED** [review]
